rerun_py/src/catalog/dataset.rs

```diff
@@ -98,11 +98,19 @@ impl PyDataset {
 
     /// Register a RRD URI to the dataset.
     fn register(self_: PyRef<'_, Self>, recording_uri: String) -> PyResult<()> {
+        // TODO(andrea): In order to make the `register` method to appear synchronous,
+        // we need to hard-code a max timeout for waiting for the task.
+        // 60 seconds is totally arbitrary but should work for now.
+        //
+        // A more permanent solution is to expose an asynchronous register method, and/or
+        // the timeout directly to the caller.
+        const MAX_REGISTER_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(60);
         let super_ = self_.as_super();
         let mut connection = super_.client.borrow(self_.py()).connection().clone();
         let dataset_id = super_.details.id;
 
-        connection.register_with_dataset(self_.py(), dataset_id, recording_uri)
+        let task_id = connection.register_with_dataset(self_.py(), dataset_id, recording_uri)?;
+        connection.wait_for_task(self_.py(), &task_id, MAX_REGISTER_TIMEOUT)
```

Comment:
I think the end game here is to immediately return some kind of `PyRemoteTask` object that would expose some task management APIs, such as `.status()` and `.block(timeout)`. Would be great if you could open an issue about that and add a todo here.